### PR TITLE
CATTY-250 Create model for UserList

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
- Created a new class `UserList` (Catty/DataModel/UserData/)
- Moved the UserVariable and UserDataProtocol to this new folder
- Removed the empty Catty/DataModel/UserVariable folder
- Added a new test  for the new class `UserList`

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/CattyTests/DataModel/UserListTest.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        context.updateReference(userListA, withReference: userListB)
+        XCTAssertEqual(1, context.updatedReferences.count)
+
+        let userListCopy = userListA.mutableCopy(with: context) as! UserList
+        XCTAssertEqual(userListA.name, userListCopy.name)
+        XCTAssertFalse(userListA === userListCopy)
+        XCTAssertTrue(userListB === userListCopy)
+    }
+
+    func testIsEqualToUserListForDifferentName() {
```
Comment (m-herold) [inline src/CattyTests/DataModel/UserListTest.swift:66]: Please also consider the actual values in your tests (like in https://github.com/Catrobat/Catty/blob/develop/src/CattyTests/DataModel/UserVariableTests.swift#L94).

File: src/Catty/DataModel/UserData/UserList.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class UserList: NSObject, UserDataProtocol {
+
+    var textLabel: SKLabelNode?
```
Comment (m-herold) [inline src/Catty/DataModel/UserData/UserList.swift:25]: Please add an explicit associated type here and remove the _textLabel_ (only relevant for variables).

File: src/CattyTests/DataModel/UserListTest.swift (hunk 1 above)
Comment (DhrubojyotiBis1) [reply src/CattyTests/DataModel/UserListTest.swift:66]: Hi @m-herold  `UserList.value` should always be a list right? how to assign `NSMutableArray` to the value of a `UserList` which is of type `[Any]?`. Also if the value is not assigned an `NSMutableArray` then the `Util.Equal` function does not work

please correct me if I am doing something wrong.

Comment (m-herold) [reply src/CattyTests/DataModel/UserListTest.swift:66]: Yes, this value should always be an array. Ok, that would mean we will have to write our own comparison function for two arrays containing either Strings or numbers. For that, I will create another ticket, please do not do this as part of this ticket. Instead, for now, change the type to _NSMutableArray_ (do not forget to add the _typealias_). Please note: change optional to non-optional, and initialize an empty array in the initializer. You will have to make one minor change to the protocol for that and add an optional _typealias_ for the _UserVariable_. Thanks!

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold  I have made the changes. please review

File: src/Catty/DataModel/UserData/UserList.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class UserList: NSObject, UserDataProtocol {
+    typealias DataType = NSMutableArray
+
+    var name: String
+    var value: NSMutableArray
+
+    override private init() {
```
Comment (m-herold) [inline src/Catty/DataModel/UserData/UserList.swift:29]: This initializer can be removed.

File: src/Catty/DataModel/UserData/UserList.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        self.value = NSMutableArray()
+    }
+
+    init(name: String) {
+        self.name = name
+        self.value = NSMutableArray()
+    }
+
+    init(list: UserList) {
+        self.name = list.name
```
Comment (m-herold) [inline src/Catty/DataModel/UserData/UserList.swift:40]: Please forward to the init with name initializer.

File: src/CattyTests/DataModel/UserListTest.swift [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        XCTAssertEqual(userListA.name, userListCopy.name)
+        XCTAssertFalse(userListA === userListCopy)
+        XCTAssertTrue(userListB === userListCopy)
+    }
+
+    func testIsEqual() {
+        let listA = UserList(name: "userList")
+        let listB = UserList(name: "userList")
+
+        listA.value = NSMutableArray(array: [50, 51])
```
Comment (m-herold) [inline src/CattyTests/DataModel/UserListTest.swift:70]: Please also add another String item to the array (e.g., "itemA").

File: src/Catty/DataModel/UserData/UserList.swift (hunk 4 above)
Comment (m-herold) [reply src/Catty/DataModel/UserData/UserList.swift:40]: Sorry, this can be ignored.

**Pull request merged**